src/hocon_tconf.erl

```diff
@@ -433,18 +410,19 @@ do_map2(Fields, Value0, Opts) ->
 
 map_fields([], Conf, Mapped, _Opts) ->
     {Mapped, Conf};
-map_fields([{FieldName, FieldSchema} | Fields], Conf0, Acc, Opts) ->
+map_fields([{_, FieldSchema} = Field | Fields], Conf0, Acc, Opts) ->
     case hocon_schema:is_deprecated(FieldSchema) of
         true ->
-            Conf = del_value(Opts, bin(FieldName), Conf0),
+            Conf = del_value(Opts, name_and_aliases(Field), Conf0),
             map_fields(Fields, Conf, Acc, Opts);
         false ->
-            map_fields_cont([{FieldName, FieldSchema} | Fields], Conf0, Acc, Opts)
+            map_fields_cont([Field | Fields], Conf0, Acc, Opts)
     end.
 
-map_fields_cont([{FieldName, FieldSchema} | Fields], Conf0, Acc, Opts) ->
+map_fields_cont([{_, FieldSchema} = Field | Fields], Conf0, Acc, Opts) ->
     FieldType = field_schema(FieldSchema, type),
-    FieldValue = get_field(Opts, FieldName, Conf0),
+    [FieldName | Aliases] = name_and_aliases(Field),
+    FieldValue = get_field_value(Opts, [FieldName | Aliases], Conf0),
```

Comment:
Also, is it right that the schema generally allows alias clashes? I see that the `hocon_schema_json` deals with this, but here it seems to be allowed? Again, worth describing in the same comment what are the implications, e.g. what alias / field would win in a clash?

Reply:
the duplication check can be done at compile time -- this is why it's moved to hocon_schema_json which is a compile-time dump.

it does not make sense to repeatedly check schema sanity at runtime.